### PR TITLE
TG-554 | Partnership multiproduct

### PR DESCRIPTION
**Pull request by diogo-wopta**

### CODE REVIEW

File: lib/jwt.go [hunk 1]
```diff
@@ -2,30 +2,57 @@ package lib
...
+		log.Printf("[DecryptJwt] could not parse jwt - %s", err.Error())
+		return nil, fmt.Errorf("could not parse jwt")
+	}
+
+	decodedKey, err := b64.StdEncoding.DecodeString(key)
 	if err != nil {
-		log.Printf("[DecryptJwt] could not decrypt jwt - %s", err.Error())
-		return fmt.Errorf("could not decrypt jwt")
+		log.Printf("[decryptJwt] could not decode signing key - %s", err.Error())
+		return nil, fmt.Errorf("could not decode jwt key")
```
Comment (diogo-wopta) [inline lib/jwt.go:54]: fix logs

File: partnership/get-node-products.go [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+	"github.com/wopta/goworkspace/lib"
+	"github.com/wopta/goworkspace/models"
+	"github.com/wopta/goworkspace/network"
+	"github.com/wopta/goworkspace/product"
+)
+
+type Response struct {
+	Partnership PartnershipNode      `json:"partnership"`
+	Products    []models.ProductInfo `json:"products"`
+}
```
Comment (diogo-wopta) [inline partnership/get-node-products.go]: rename me

File: models/partnership-node.go [hunk 3]
```diff
@@ -1,12 +1,63 @@
...
+	Phone      string `json:"phone"`
+	Address    string `json:"address"`
+	Postalcode string `json:"postalCode"`
+	City       string `json:"city"`
+	CityCode   string `json:"cityCode"`
+	Work       string `json:"work"`
+	Guarantees map[string]struct {
+		Duration                   int
+		SumInsuredLimitOfIndemnity float64
+	} `json:"guarantees"`
```
Comment (Yousef-Hammar) [inline models/partnership-node.go]: add missing `json` tag

File: partnership/get-node-products.go [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+		node     *models.NetworkNode
+		err      error
+	)
+
+	log.SetPrefix("[GetPartnershipNodeAndProductsFx] ")
+	defer log.SetPrefix("")
+
+	log.Println("Handler start -----------------------------------------------")
+
+	affinity := chi.URLParam(r, "partnershipUid")
```
Comment (Yousef-Hammar) [inline partnership/get-node-products.go]: rename `affinity` in `partnershipUid`

File: partnership/life.go [hunk 5]
```diff
@@ -14,30 +15,85 @@ import (
...
+		response        PartnershipResponse
+		partnershipNode *models.NetworkNode
+		policy          models.Policy
+		productLife     *models.Product
+		claims          models.LifeClaims
+		err             error
+	)
 
-	log.SetPrefix("[LifePartnershipFx] ")
+	log.SetPrefix("[LifePartnershipV2Fx] ")
```
Comment (Yousef-Hammar) [inline partnership/life.go]: fix me

File: partnership/life.go [hunk 6]
```diff
@@ -14,30 +15,85 @@ import (
...
+
+	log.Printf("loading latest life product")
+	productLife = product.GetLatestActiveProduct(models.LifeProduct, models.ECommerceChannel, partnershipNode, nil)
+	if productLife == nil {
+		log.Printf("no product found")
+		return "", nil, fmt.Errorf("no product found")
+	}
+	policy = setPolicyPartnershipInfo(policy, productLife, partnershipNode)
+
+	if claims, err = partnershipNode.Partnership.DecryptJwtClaims(jwtData, os.Getenv(key), LifeClaimsExtractor(partnershipUid)); err != nil {
```
Comment (Yousef-Hammar) [inline partnership/life.go]: `LifeClaimsExtractor` should be private

**Pull request merged**